org-timeline.el

```diff
@@ -146,27 +142,56 @@ Return new copy of STRING."
                             (round (+ beg duration))
                           current-time))
                    (face (org-timeline--get-face)))
-              (when (>= beg start-offset)
-                (push (list beg end face) tasks)))))))
-    (setq tasks (nreverse tasks))
+              (push (list beg end face) tasks))))))
+    (nreverse tasks)))
+
+(defun org-timeline--generate-hourline ()
+  "Generate the template for the timeline's hourline."
+  (let* ((hourline "|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|  :00|")
+         (start-hour (abs org-timeline-start-hour))
+         (current-slot 0))
+    (while (< current-slot 24)
+      (let* ((current-slot-hour (mod (+ start-hour current-slot) 24))
+            (current-slot-ones (mod current-slot-hour 10))
+            (current-slot-tens (/ (- current-slot-hour current-slot-ones) 10)))
+        (aset hourline (+ 1 (* current-slot 6)) (+ 48 current-slot-tens)) ;; 48 is ascii for '0'
+        (aset hourline (+ 2 (* current-slot 6)) (+ 48 current-slot-ones)))
+      (cl-incf current-slot))
+    hourline))
```

Comment:
Usually using `aset` and similar functions which modify by side effect is not the common approach in lisp.  More lispy/functional approach would be to generate each hour slot and then join them into one string.

A possible implementation could be something like this.  Of course there are some additional changes needed, like the wrapping `|` and customizable start (currently I just put 5 there).

```elisp
(mapconcat (lambda (x) (format "%02d:00" (mod x 24))) (number-sequence 5 (+ 5 23)) "|")
```

This works by first creating a number sequence, then applying the lambda on each item of the list and finally joining all the strings with `"|"`.

It could also be implemented in two steps, first using `mapcar` to generate the list:

```elisp
(mapcar (lambda (x) (format "%02d:00" (mod x 24))) (number-sequence 5 (+ 5 23)))
```

and then join it into one string

```elisp
(string-join list-of-hours "|")
```

Reply:
Thank you for the tips! I'll implement it that way.